rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -90,44 +104,100 @@ get_node_names(
     if (DDS::RETCODE_OK == dds_ret) {
       auto data = static_cast<unsigned char *>(pbtd.user_data.value.get_contiguous_buffer());
       std::vector<uint8_t> kv(data, data + pbtd.user_data.value.length());
+
       auto map = rmw::impl::cpp::parse_key_value(kv);
+
       auto name_found = map.find("name");
       auto ns_found = map.find("namespace");
 
       if (name_found != map.end()) {
         name = std::string(name_found->second.begin(), name_found->second.end());
       }
-      if (name_found != map.end()) {
+
+      if (ns_found != map.end()) {
         namespace_ = std::string(ns_found->second.begin(), ns_found->second.end());
       }
+      
       if (name.empty()) {
         // use participant name if no name was found in the user data
         if (pbtd.participant_name.name) {
           name = pbtd.participant_name.name;
         }
       }
     }
+
+    // ignore discovered participants without a name
     if (name.empty()) {
-      // ignore discovered participants without a name
-      node_names->data[i] = nullptr;
-      node_namespaces->data[i] = nullptr;
+      tmp_names_list.data[i] = nullptr;
+      tmp_namespaces_list.data[i] = nullptr;
       continue;
     }
 
-    node_names->data[i] = rcutils_strdup(name.c_str(), allocator);
-    if (!node_names->data[i]) {
+    tmp_names_list.data[named_nodes_num] = rcutils_strdup(name.c_str(), allocator);
+    if (!tmp_names_list.data[named_nodes_num]) {
       RMW_SET_ERROR_MSG("could not allocate memory for node name");
       goto fail;
     }
 
-    node_namespaces->data[i] = rcutils_strdup(namespace_.c_str(), allocator);
-    if (!node_namespaces->data[i]) {
+    ++named_nodes_num;
+    
+    if (namespace_.empty()) {
+      tmp_namespaces_list.data[i] = nullptr;
```

Comment:
This is concerning, since we expect that the output names and namespaces lists will be the same size representing a one-to-one mapping for nodes ([docs](https://github.com/ros2/rmw/blob/813b94ddd5650444b312304ad156f3b5d9f05e39/rmw/include/rmw/rmw.h#L1015-L1018)). It is possible that if we have empty namespaces, but not empty names, that we will end up with lists of different sizes and lose the mapping. In other words, we want to ensure that `named_nodes_num == named_namespaces_num`.

I think that if the namespace is empty, we can let it be empty and not filter it like we do with the node names. We can probably just remove this block.

Reply:
In fact, we might remove the variable `named_namespaces_num` altogether since it should always be the same as `named_nodes_num`.

Reply:
Done